ALUM/ALUM/CustomErrors.swift

```diff
@@ -6,10 +6,22 @@
 //
 
 import Foundation
+import FirebaseAuth
 
-enum APIError: Error {
-    case networkError(message: String = "")
-    case serverError(message: String = "")
-    case authenticationError(message: String = "")
-    case invalidRequest(message: String = "")
+enum AppError: Error {
+    case actionable(ActionableError, message: String = "")
+    case internalError(InternalError, message: String = "")
+}
+
+enum ActionableError: Error {
```

Comment:
Streamlining error handling for our app. 

Actionable Errors are ones that we can show to the user and they can do something about it. For eg. no network. 

Internal Errors are ones that happened because of something we coded wrong. We are not supposed to have internal errors in the published app but during the testing phase it will be useful to log them